simulation/traffic_simulator/include/traffic_simulator/entity/ego_entity.hpp

```diff
@@ -25,57 +25,22 @@
 #include <string>
 #include <traffic_simulator/api/configuration.hpp>
 #include <traffic_simulator/entity/vehicle_entity.hpp>
-#include <traffic_simulator/helper/helper.hpp>
-#include <traffic_simulator/vehicle_model/sim_model.hpp>
+#include <traffic_simulator/vehicle_simulation/vehicle_model/sim_model.hpp>
 #include <traffic_simulator_msgs/msg/entity_type.hpp>
 #include <vector>
 
-template <typename T>
-auto getParameter(const std::string & name, T value = {})
-{
-  rclcpp::Node node{"get_parameter", "simulation"};
-
-  node.declare_parameter<T>(name, value);
-  node.get_parameter<T>(name, value);
-
-  return value;
-}
-
 namespace traffic_simulator
 {
 namespace entity
 {
-enum class VehicleModelType {
-  DELAY_STEER_ACC,
-  DELAY_STEER_ACC_GEARED,
-  DELAY_STEER_VEL,
-  IDEAL_STEER_ACC,
-  IDEAL_STEER_ACC_GEARED,
-  IDEAL_STEER_VEL,
-};
-
 class EgoEntity : public VehicleEntity
 {
   const std::unique_ptr<concealer::FieldOperatorApplication> field_operator_application;
-  const std::unique_ptr<concealer::Autoware> autoware;
-
-  const VehicleModelType vehicle_model_type_;
-
-  const std::shared_ptr<SimModelInterface> vehicle_model_ptr_;
-
-  std::optional<geometry_msgs::msg::Pose> initial_pose_;
-
-  std::optional<double> previous_linear_velocity_, previous_angular_velocity_;
-
-  static auto getVehicleModelType() -> VehicleModelType;
 
   static auto makeFieldOperatorApplication(const Configuration &)
     -> std::unique_ptr<concealer::FieldOperatorApplication>;
 
-  static auto makeSimulationModel(
-    const VehicleModelType, const double step_time,
-    const traffic_simulator_msgs::msg::VehicleParameters &)
-    -> const std::shared_ptr<SimModelInterface>;
+  traffic_simulator_msgs::msg::EntityStatus externally_updated_status_;
```

Comment:
Why this member valus is needed?
Please explain why we need to create a new `externally_updated_status_` and not rewrite the `status_` directly.

Reply:
In `EgoEntity::onUpdate()` is called `EntityBase::OnUpdate()` and then `EgoEntity::setStatus()` . `EntityBase::onUpdate()` updates `status_before_udpate_` and if `setStatus()` is called from outside then `status_before_udpate_` would be a status after the update. For that reason `setStatus(externally_updated_status_);` is called in `EgoEntity::onUpdate()`  to set the state properly.

Reply:
I think this problem should be solved by configure the update timing of the `status_before_udpate_` member value.
And also, if you want add `externally_updated_status_` member value it is not a small change, so please leave documentation in [`docs/developer_guide` directory.](https://github.com/tier4/scenario_simulator_v2/tree/24617d21fb0a221eaf67a6bee7a53d0eeebff69a/docs/developer_guide)

Reply:
I would like to let you know, I'm working on a design of a possible solution to this problem. Please give me some time.

Reply:
@hakuturu583 
During my analysis, I've noticed that `getEntityStatusBeforeUpdate()` method and `status_before_update_` member value is used only by EgoEntity [here](https://github.com/tier4/scenario_simulator_v2/blob/master/simulation/traffic_simulator/src/api/api.cpp#L282C1-L292C4) to send `UpdateEntityStatusRequest` by zeromq interfece. On `SimpleSensorSimulator` side [here](https://github.com/tier4/scenario_simulator_v2/blob/master/simulation/simple_sensor_simulator/src/simple_sensor_simulator.cpp#L106C1-L117C2) the information about `ego_entity_status_before_update` is not used.
Moreover, In the [new ZMQ API](https://docs.google.com/document/d/1lQIVPp-eHlSMdf8ocGOOIWrZQRSbBYOB6BIkStUJEbo/edit) the `ego_entity_status_before_update` is not supposed to be used.

I would like to ask if do know in which cases the  `status_before_update_` member value is supposed to be use? 
I think that the `status_before_update_` is obsolete.  I'm not sure if I miss something.

Reply:
Internally discussed in our team and we agree with `status_before_update_` will be unnecessary when the ego entity was moved to `simple_sensor_simulator`.
I will accept the current implementation on the condition that the code is simplified by removing `status_before_updated`.